libethash-cuda/CUDAMiner.cpp

```diff
@@ -113,75 +115,87 @@ void CUDAMiner::report(uint64_t _nonce)
 void CUDAMiner::kickOff()
 {
 	m_hook->reset();
-	startWorking();
+	//startWorking();
 }
 
-void CUDAMiner::workLoop()
+bool CUDAMiner::init(const h256& seed)
 {
 	// take local copy of work since it may end up being overwritten by kickOff/pause.
 	try {
-		WorkPackage w = work();
-		if (!w)
-			return;
+		unsigned device = s_devices[index] > -1 ? s_devices[index] : index;
 
-		if (!m_miner || m_minerSeed != w.seed)
-		{
-			unsigned device = s_devices[index] > -1 ? s_devices[index] : index;
+		cnote << "Initialising miner...";
+		m_minerSeed = seed;
 
-			if (s_dagLoadMode == DAG_LOAD_MODE_SEQUENTIAL)
-			{
-				while (s_dagLoadIndex < index) {
-					this_thread::sleep_for(chrono::seconds(1));
-				}
-			}
-			else if (s_dagLoadMode == DAG_LOAD_MODE_SINGLE)
-			{
-				if (device != s_dagCreateDevice)
-				{
-					// wait until DAG is created on selected device
-					while (s_dagInHostMemory == NULL) {
-						this_thread::sleep_for(chrono::seconds(1));
-					}
-				}
-				else
-				{
-					// reset load index
-					s_dagLoadIndex = 0;
-				}
-			}
-
-			cnote << "Initialising miner...";
-			m_minerSeed = w.seed;
-
-			delete m_miner;
+		//delete m_miner;
+		//m_miner = new ethash_cuda_miner;
+		if(!m_miner)
```

Comment:
You can just embed ethash_cuda_miner in CUDAMiner. I will remove it anyway later on.

Reply:
I don't know what you mean

Reply:
Nevermind, I will fix it later.